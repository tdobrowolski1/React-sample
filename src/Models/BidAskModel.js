class BidAskModel {
    Value;
    Lot;
    
}

export default BidAskModel