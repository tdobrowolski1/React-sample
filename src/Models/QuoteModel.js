class QuoteModel {
    Ask;
    Bid;
    Ticker;
}

export default QuoteModel;