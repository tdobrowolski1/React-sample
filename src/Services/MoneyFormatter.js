
class MoneyFormater {
    //mocking it for sample EURUSD scenario
    FormatBeginPrice(price) {
        return price.toString().substring(0,1) + "." +price.toString().substring(1,3) ;

    }
    FormatMidPrice(price){
        return price.toString().substring(3,5)
    }

    FormatEndPrice(price){
        return price.toString().substring(5,6) ;
    }

}

export default MoneyFormater;