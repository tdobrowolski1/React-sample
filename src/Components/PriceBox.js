import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import '../App.css';
import QuoteModel from '../Models/QuoteModel'
import MoneyFormatter from '../Services/MoneyFormatter'
class PriceBox extends PureComponent {
    render() {
        const price = this.props.isAsk ? this.props.quote.Ask.Value : this.props.quote.Bid.Value;
        return (
            <div className={"PriceBox " + (this.props.isAsk ? 'ask' : 'bid')} onClick={(e) => this.props.onClickCallback(e,this.props.isAsk, this.props.quote)} >
                <div className={"PriceBoxHeader " + (this.props.isAsk ? 'floatRight' : 'floatLeft')}>
                    {this.props.quote.Ticker}
                </div>
                <div className={"PriceBoxHeader " + (this.props.isAsk ? 'floatLeft' : 'floatRight')}>
                    {this.props.moneyFormatter.FormatBeginPrice(this.props.quote.Ask.Value)}
                </div>
                <div className="PriceBoxBody">
                    <span className="mediumPrice"> {this.props.moneyFormatter.FormatMidPrice(price)}</span> <span className="smallPrice">{this.props.moneyFormatter.FormatEndPrice(price)}</span>
                </div>
            </div >
        );
    }
}
PriceBox.propTypes = {
    isAsk: PropTypes.bool,
    quote: PropTypes.instanceOf(QuoteModel),
    moneyFormatter : PropTypes.instanceOf(MoneyFormatter),
    onClickCallback : PropTypes.func,
}

export default PriceBox;
