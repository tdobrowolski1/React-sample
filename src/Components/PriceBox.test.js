import React from 'react';
import PriceBox from './PriceBox'
import renderer from 'react-test-renderer';
import MoneyFormatter from '../Services/MoneyFormatter';
import QuoteModel from '../Models/QuoteModel'
import BidAskModel from '../Models/BidAskModel'
import { shallow, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });



describe('PriceBox', () => {
    var onClickCallback;
    var moneyFormatter;
    var quote;
    beforeEach(() => {
        onClickCallback = jest.fn();
        moneyFormatter = new MoneyFormatter();
        quote = new QuoteModel();
        quote.Ticker = "EURUSD";
        quote.Bid = new BidAskModel();
        quote.Ask = new BidAskModel();
        quote.Bid.Value = 123456;
        quote.Ask.Value = 123456;
    })
    it('renders correctly', () => {
        const ask = renderer
            .create(<PriceBox onClickCallback={onClickCallback} quote={quote} moneyFormatter={moneyFormatter} isAsk={true} />)
            .toJSON();
        expect(ask).toMatchSnapshot();
        const bid = renderer
            .create(<PriceBox onClickCallback={onClickCallback} quote={quote} moneyFormatter={moneyFormatter} isAsk={false} />)
            .toJSON();
        expect(bid).toMatchSnapshot();
    });
    it('invokes callback when clicked on', () => {
        const priceBox = shallow(<PriceBox onClickCallback={onClickCallback} quote={quote} moneyFormatter={moneyFormatter} isAsk={false} />);
        priceBox.find('.PriceBox').simulate('click');
        expect(onClickCallback).toBeCalled();
    });

});