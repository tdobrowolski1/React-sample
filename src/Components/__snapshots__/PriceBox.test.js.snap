// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceBox renders correctly 1`] = `
<div
  className="PriceBox ask"
  onClick={[Function]}
>
  <div
    className="PriceBoxHeader floatRight"
  >
    EURUSD
  </div>
  <div
    className="PriceBoxHeader floatLeft"
  >
    1.23
  </div>
  <div
    className="PriceBoxBody"
  >
    <span
      className="mediumPrice"
    >
       
      45
    </span>
     
    <span
      className="smallPrice"
    >
      6
    </span>
  </div>
</div>
`;

exports[`PriceBox renders correctly 2`] = `
<div
  className="PriceBox bid"
  onClick={[Function]}
>
  <div
    className="PriceBoxHeader floatLeft"
  >
    EURUSD
  </div>
  <div
    className="PriceBoxHeader floatRight"
  >
    1.23
  </div>
  <div
    className="PriceBoxBody"
  >
    <span
      className="mediumPrice"
    >
       
      45
    </span>
     
    <span
      className="smallPrice"
    >
      6
    </span>
  </div>
</div>
`;
